docs-ui/stories/components/profiling/flamegraphZoomView.stories.js

```diff
@@ -3,19 +3,47 @@ import {FullScreenFlamegraphContainer} from 'sentry/components/profiling/FullScr
 import {FlamegraphPreferencesProvider} from 'sentry/utils/profiling/flamegraph/FlamegraphPreferencesProvider';
 import {FlamegraphThemeProvider} from 'sentry/utils/profiling/flamegraph/FlamegraphThemeProvider';
 import {importProfile} from 'sentry/utils/profiling/profile/importProfile';
+import {JSSelfProfile} from 'sentry/utils/profiling/profile/jsSelfProfile.tsx';
+import {createFrameIndex} from 'sentry/utils/profiling/profile/utils';
 
 export default {
   title: 'Components/Profiling/FlamegraphZoomView',
 };
 
-const profiles = importProfile(require('./EventedTrace.json'));
+const eventedProfiles = importProfile(require('./EventedTrace.json'));
 
 export const EventedTrace = () => {
   return (
     <FlamegraphPreferencesProvider>
       <FlamegraphThemeProvider>
         <FullScreenFlamegraphContainer>
-          <Flamegraph profiles={profiles} />
+          <Flamegraph profiles={eventedProfiles} />
+        </FullScreenFlamegraphContainer>
+      </FlamegraphThemeProvider>
+    </FlamegraphPreferencesProvider>
+  );
+};
+
+const jsSelfProfile = require('./JSSelfProfilingTrace.json');
+
+export const JSSelfProfiling = () => {
+  const profiles = {
+    name: 'JS Self Profiling',
+    activeProfileIndex: 0,
+    traceID: '',
+    profiles: [
+      JSSelfProfile.FromProfile(
+        jsSelfProfile,
+        createFrameIndex(jsSelfProfile.frames, jsSelfProfile)
+      ),
+    ],
+  };
```

Comment:
Is there a reason not to use `importProfile` here?

Reply:
Yeah, the importProfile right now accepts the full format (the trace here is just a dump from the window.profiler). But we can use importDroppedProfile, so let me change that

Reply:
Actually, no. Right now importDroppedProfile accepts a File, then we do the same as here
```
if (isJSProfile(json)) {
          return {
            name: 'JS Self Profiling',
            activeProfileIndex: 0,
            traceID: '',
            profiles: [JSSelfProfile.FromProfile(json, createFrameIndex(json.frames))],
          };
        }
```